engine/Shopware/Controllers/Backend/Notification.php

```diff
@@ -32,22 +33,21 @@ class Shopware_Controllers_Backend_Notification extends Shopware_Controllers_Bac
     public function getArticleListAction()
     {
         try {
-            $limit = (int) $this->Request()->limit;
-            $offset = (int) $this->Request()->start;
+            $limit = (int) $this->Request()->get('limit', 20);
+            $offset = (int) $this->Request()->get('start', 0);
 
             /** @var array $filter */
             $filter = $this->Request()->getParam('filter', []);
 
             // order data
             $order = (array) $this->Request()->getParam('sort', []);
 
+            /** @var ArticleRepository $repository */
             $repository = $this->get('models')->getRepository(Article::class);
             $dataQuery = $repository->getArticlesWithRegisteredNotificationsQuery($filter, $offset, $limit, $order);
-            $data = $dataQuery->getArrayResult();
 
-            // manually calc the totalAmount cause the paginate($this->get('models')->getQueryCount) doesn't work with this query
-            $dataQuery->setFirstResult(0);
-            $totalCount = \count($dataQuery->getArrayResult());
+            $data = $dataQuery->getResult(AbstractQuery::HYDRATE_ARRAY);
+            $totalCount = $repository->getArticleWithNotificationCountQuery()->getOneOrNullResult(AbstractQuery::HYDRATE_SINGLE_SCALAR);
```

Comment:
```suggestion
            $totalCount = (int) $repository->getArticleWithNotificationCountQuery()->getOneOrNullResult(AbstractQuery::HYDRATE_SINGLE_SCALAR);
```

do the cast here and not where the variable is assigned to another one